src/v2/ion/uiSchemaTransformer.js

```diff
@@ -70,15 +70,30 @@ const createUISchema = (remediationValue = []) => {
   });
 };
 
+const addFactorType = (options, factors) => {
+  _.each(options, function (optionItem) {
+    const factorValue = optionItem.value;
+    const factor = factors.find(function (item) {
+      return (item.factorProfileId === factorValue
+        || item.factorId === factorValue);
+    });
+    optionItem.factorType = factor.factorType;
+  });
+};
+
 /**
  *
  * @param {AuthResult} transformedResp (after `actionsTransformer`)
  */
 const insertUISchema = (transformedResp) => {
   if (transformedResp.currentState) {
     const remediation = transformedResp.currentState.remediation || [];
+    const factors = transformedResp.factors && transformedResp.factors.value || [];
     transformedResp.currentState.remediation = remediation.map(obj => {
       obj.uiSchema = createUISchema(obj.value);
+      if (obj.value && obj.value[0] && obj.value[0].options) {
```

Comment:
`value[0]` is too specific. why not adding `factorType` (invokes addFactorType) around line 63?
```
    if (ionFormField.name === 'factorId' ||
      ionFormField.name === 'factorProfileId') {
      uiSchema.type = 'factorType';
    }
```

Reply:
yeah, since `createUISchema` method is dealing with the remediation object to create the UI Schema I was trying to avoid sending in the factors object into that just for this.

Reply:
I could just loop through each of the items in `obj.value`

``` 
_.each(obj.value, function (optionsObject) {
  if (optionsObject.options) {
    addFactorTypeToOptions(optionsObject.options, factors);
  }
});
  
```

Reply:
> since createUISchema method is dealing with the remediation object to create the UI Schema

It's true before but not now which says 'in order to create ui schema, we need remediation and factors`